app/controllers/users/forgot-password.js

```diff
@@ -2,33 +2,20 @@ import { inject as service } from '@ember/service';
 import Controller from '@ember/controller';
 
 export default Controller.extend({
-  ajax: service(),
+  fetch: service(),
 
   queryParams: ['email'],
   email: null,
 
   actions: {
-    forgotPassword() {
-      const resetPasswordRequest = this.ajax.post('/users/reset_password', { data: { email: this.email } });
-      resetPasswordRequest.then(() => {
-        const successMessage = 'Je ontvangt een e-mail met verdere instructies';
-        this.set('successMessage', successMessage);
-      }, reject => {
-        let errorMessage;
-        switch (reject.status) {
-          case 404:
-            errorMessage = 'E-mailadres niet gevonden';
-            break;
-          case 500:
-            errorMessage = ('Er ging iets mis op de server, <a href="mailto:ict@csvalpha.nl">raadpleeg de ICT-commissie</a>').htmlSafe();
-            break;
-          default:
-            errorMessage = 'Er ging iets mis, probeer het nog een keer';
-            break;
-        }
+    async forgotPassword() {
+      const response = await this.fetch.post('/users/reset_password', { body: { email: this.email } });
 
-        this.set('errorMessage', errorMessage);
-      });
+      if (response.ok) {
+        this.set('successMessage', 'Je ontvangt een e-mail met verdere instructies');
+      } else {
```

Comment:
Should ember-fetch/errors' isNotFoundResponse and isServerErrorResponse be used here to preserve the old error messages?

Reply:
If we check the code at the backed it will never return a 404 (which it also shouldn't since this gives information about the existing users). So that line of code was never being touched. 

https://github.com/csvalpha/amber-api/blob/staging/app/controllers/v1/users_controller.rb#L50

Reply:
Oh okay, that makes sense!